content/en/docs/Examples/kubeadm-provider.md

```diff
@@ -0,0 +1,191 @@
+---
+title: A Minimal Single-Node Kubernetes with Kubeadm
+linkTitle: A Minimal Single-Node Kubernetes with Kubeadm
+description: Learn how to build a Kairos image for a single-node Kubernetes cluster using the provider-kubeadm.
+---
+
+
+Kairos is all about giving you the power to customize your operating system just the way you need it—declaratively, reproducibly, and predictably. Today, we're walking through how to build and boot a Kairos image using the [provider-kubeadm](https://github.com/kairos-io/provider-kubeadm) to set up a Kubernetes cluster with `kubeadm`.
+
+This guide is focused on a simple use case: booting a **single-node Kubernetes cluster** with role `init`, version `v1.30.0`.
+
+---
+
+## 🧱 What Is `provider-kubeadm`?
+
+The `provider-kubeadm` is an [init provider](https://kairos.io/docs/reference/init-providers/) for Kairos that integrates with Kubernetes' `kubeadm` bootstrap process. It translates the familiar `kubeadm` configuration into a Kairos-compatible cloud-init YAML, wrapping everything in a reproducible and declarative boot process.
+
+With this provider, we can fully define Kubernetes cluster parameters—including API server args, scheduler, networking, and etcd configuration—right inside a Kairos `#cloud-config` block.
+
+---
+
+## 🔧 Building the Image
+
+We start with a Kairos base image—here, Ubuntu 24.04 Core—and layer on everything `kubeadm` needs: containerd, kubelet, kubectl, and the kubeadm binary. Also we get the agent-provide-kubeadm to handle the `kubeadm` configuration.
+
+Here's the Dockerfile used to construct the image:
+
+```Dockerfile
+FROM quay.io/kairos/ubuntu:24.04-core-amd64-generic-v3.4.2
```

Comment:
Shall we have an example with kairos-init instead? We want to drive away from our images in the future (to avoid having large matrix builds), so better start referencing it IMHO